source/client/process_impl.cc

```diff
@@ -336,11 +337,10 @@ void ProcessImpl::createBootstrapConfiguration(envoy::config::bootstrap::v3::Boo
     cluster->set_name(fmt::format("{}", i));
     cluster->mutable_connect_timeout()->set_seconds(options_.timeout().count());
     cluster->mutable_max_requests_per_connection()->set_value(options_.maxRequestsPerConnection());
-    if (options_.h2()) {
+    if (options_.upstreamProtocol() == Envoy::Http::Protocol::Http2) {
```

Comment:
There will as you have probably seen by now in https://github.com/envoyproxy/nighthawk/pull/723.

Since this code changes significantly in this PR chain, it probably isn't worth adding another throw here. Support for Http3 will be dully implemented before this PR gets merged into master.

Feel free to reopen if you still believe this is required here.